ts/src/index.ts

```diff
@@ -129,13 +130,28 @@ export async function start(config: Config = {}): Promise<void> {
         config);
     return;
   }
+  // start the v8 heap profiler if heap profiling is enabled. This must be done
+  // before any asynchronous calls are made so samples from the first tick can
+  // be captured.
+  if (!(config.disableHeap === false)) {
+    heapProfiler.startSamplingHeapProfiler(
+        config.heapIntervalBytes || defaultConfig.heapIntervalBytes,
+        config.heapMaxStackDepth || defaultConfig.heapIntervalBytes);
+  }
+
   let normalizedConfig: ProfilerConfig;
   try {
     normalizedConfig = await initConfig(config);
   } catch (e) {
+    heapProfiler.stopSamplingHeapProfiler();
```

Comment:
I think you want it in finally{}, not in catch{}. Then you can have it just once.